HIP/hip-866.md

```diff
@@ -0,0 +1,149 @@
+---
+hip: 866
+title: Add EVM compatibility for non-supported Cancun blob features.
+author: Danno Ferrin (@shemnon)
+working-group: Nana Essilfie-Conduah <@nana-ec>, Jasper Potts <@jasperpotts>, Richard Bair <@rbair23>
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Draft
+created: 2024-01-24
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/872
+---
+
+## Abstract
+
+Add compatibility for Ethereum Mainnet Cancun features relating to Hedera's
+non-support of EIP-4844 blobs.
+
+## Motivation
+
+In order to maintain maximum compatibility with Ethereum Mainnet there will need
+to be support for some Blob opcodes in the EVM that behave appropriately in the
+absence of Blobs support in Hedera. At the same time, space needs to be
+preserved in the event Hedera provides similar Blob functionality that needs to
+be mapped to the [EIP-4844](https://eips.ethereum.org/EIPS/eip-4844) facilities.
+
+## Rationale
+
+In order to both preserve compatibility and future design space we need to act
+as if blobs are not being added to Hedera. This allows existing contracts that
+may depend on blob behavior to continue to function in the absence of blobs.
+
+First we will prevent blobs from entering the system. There are complex EL/CL
+interactions in Ethereum Mainnet, but as far as Hedera's consensus is similar to
+Ethereum Mainnet's, the main entry point is a "Type 3" transaction that allows
+blobs to be attached. Furthermore, Type 3 transactions require a blob to be
+attached. Prohibiting Type 3 transactions will thus be sufficient to keep blobs
+form entering the EVM's realm of concern and similarly will not prohibit other
+interactions that are desirable to Hedera.
+
+Second, we need to support the new opcodes but ensure they operate as they would
+in the absence of any Blobs.
+
+## User stories
+
+* As a smart contract deployer, I want my smart contract that expects blob
+  support to not fail in unexpected ways.
+* As a smart contract user, I will not be able to successfully use contracts
```

Comment:
not really a user story, can drop unless you can state it in an affirmative form of what a user should be able to do